controlplane/balancer.cpp

```diff
@@ -930,8 +936,152 @@ void balancer_t::reconfigure_wlc_thread()
 {
 	while (!flagStop)
 	{
-		balancer_real_flush();
+		if (balancer_t::reconfigure_wlc())
+		{
+			balancer_real_flush();
+		}
 
 		std::this_thread::sleep_for(std::chrono::seconds(YANET_CONFIG_BALANCER_WLC_RECONFIGURE));
 	}
 }
+
+bool balancer_t::reconfigure_wlc()
+{
+	bool wlc_weight_changed = false;
+
+	common::idp::updateGlobalBaseBalancer::request balancer;
+	const auto balancer_real_connections = dataplane.balancer_real_connections();
+
+	std::lock_guard<std::mutex> guard(reals_enabled_mutex);
+
+	for (const auto& [module_name, balancer] : generations_config.current().config_balancers)
+	{
+
+		for (const auto& [service_id,
+		                  virtual_ip,
+		                  proto,
+		                  virtual_port,
+		                  version,
+		                  scheduler,
+		                  scheduler_params,
+		                  forwarding_method,
+		                  flags,
+		                  ipv4_outer_source_network,
+		                  ipv6_outer_source_network,
+		                  reals] : balancer.services)
+		{
+			(void)flags;
+			(void)version;
+			(void)forwarding_method;
+			(void)ipv4_outer_source_network;
+			(void)ipv6_outer_source_network;
```

Comment:
Recently I've merged my cleanup pr, so now we should use `YANET_GCC_BUG_UNUSED` macro instead of void cast to do this. You can check yourself:

> commit 01422b7a7fd6787dfb44288a08c17d6a74d70811
> Author: Ivan Morozko <ivan.morozko01@gmail.com>
> Date:   Wed Sep 25 18:54:52 2024 +0400
> 
>     Mark unused structured bindings with YANET_GCC_BUG_UNUSED instead of void cast
>     
>     GCC 7.5 version which we have to support for now due to Ubuntu 18 have a bug
>     (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81767) where unused variables in
>     structured bindings are incorrectly flagged as warnings, even though they
>     "can't" be used, cause they are not a real variables.
>     
>     Current code uses `(void)var;` to suppress these warnings, but this approach
>     is ugly and harder to grep.
>     
>     This commit removes `(void)var;` statements and replaces it with `YANET_GCC_BUG_UNUSED`
>     
>     Once we move to a newer GCC version where this bug is fixed,
>     we can easily remove the that macro as it is much easier to grep and remove
>     than those void(var) statements